### DISCUSSION TITLE
Filling holes on mesh with conditions

### DISCUSSION THREAD

**Discussion by kathanvyas** [cnr-isti-vclab/meshlab#824] [Q&A]
So I have a mesh that I created using my registration algorithm. This mesh is of a dummy kid. Because the hands and legs are too close to the body, the scanning cam was not able to figure out the depth in between. so I used meshlab to remove points between the hand and body and between the mesh after I apply the screened poison. However, it leaves holes. I cannot fill the holes. Please see the figures for more details.

Image1: This image shows how there is some material in between hands and body and between two legs:
![before](https://user-images.githubusercontent.com/30324812/98862346-bd6f0100-2434-11eb-823a-6600be189ad7.JPG)

image2: This image shows after i have removed the material from the middle. this is what i want so that i can have hands and legs separate.
![after](https://user-images.githubusercontent.com/30324812/98862511-058e2380-2435-11eb-851c-c0758c2fd31a.JPG)

image3: this is my issue. After removing, it leaves such holes. when i use fill holes option, it again takes be back to something like image 1.
![holes](https://user-images.githubusercontent.com/30324812/98862531-0d4dc800-2435-11eb-9b0e-9e39ddd8a9b0.JPG)

How can I just fill these holes so I can get complete mesh without anything in the middle of the body -hands-and legs. I tried re-computing surface normals and trying again but it doesn't work.

**Comment by kathanvyas** [comment]
@alemuntoni anyway you can suggest me what to do here

**Comment by alemuntoni** [comment]
Sorry, I think there is no way to do this with the current meshlab tools.

**Comment by jmespadero** [comment]
You can do it easily, but requires some manual work.

1. Use selection tool to select whole left arm triangles. Just draw a box that contains the left arm and does not contain triangles in the torso. If you are working with more complex shapes that can't be selected with a box, just select one hand and press `ctrl-+` to invoque the `dilate selection` filter several times until you have selected the whole arm. 
2. Launch the `Close holes` filter and check the `Close holes with selected faces` option. Maybe you also have to increase the number in `max size to be closed` to be greater than default. 
3. Repeat the same process for the other arm, both legs and torso.

**Reply by kathanvyas** [reply]
Hey @jmespadero , I think I tried doing that however it will keep on adding mesh over the complete hole. including connecting hands and legs again. could you provide me a video f what you are telling so I can understand properly. I would appreciate it

**Reply by jmespadero** [reply]
You are right. In this case the boundary of the hole is cutted by the selection area, so half of the hole is outside of the selection and meshlab won't close it.
You can try another approach, but still not sure if will work. Try to use the `Transform: Scale, Normalize` filter to apply a big scale only on X axis (use a power of 2 as scale factor. So 8, 16 or 32 are good values). Dont worry about the big deformation your mesh will suffer.

![image](https://user-images.githubusercontent.com/6451497/111064647-7877db80-84b5-11eb-94a9-59110eda9f60.png)

Now, apply the `Close holes` filter and see if the triangles are sewed in the desired orientation. Apply again the  `Transform: Scale, Normalize` filter to undo the deformation in the X axis (apply 1/8, etc...)
